skbio/sequence/_grammared_sequence.py

```diff
@@ -22,11 +23,71 @@
 from ._sequence import Sequence
 
 
-class IUPACSequence(with_metaclass(ABCMeta, Sequence)):
-    """Store biological sequence data conforming to the IUPAC character set.
+class GrammaredSequenceException(Exception):
+    pass
+
+
+class GrammaredSequenceMeta(ABCMeta, type):
+    def __new__(mcls, name, parents, dct):
+        cls = super(GrammaredSequenceMeta, mcls).__new__(mcls, name, parents,
+                                                         dct)
+
+        if cls.default_gap_char not in cls.gap_chars:
+            raise GrammaredSequenceException(
+                "default_gap_char must be in gap_chars for class %s" %
+                name)
+
+        # Only perform validation on classes that aren't abstract.
+        if (type(cls.degenerate_map) is not abstractproperty and
+                type(cls.nondegenerate_chars) is not abstractproperty):
+            for key in cls.degenerate_map.keys():
+                for nondegenerate in cls.degenerate_map[key]:
+                    if nondegenerate not in cls.nondegenerate_chars:
+                        raise GrammaredSequenceException(
+                            "degenerate_map must expand only to "
+                            "characters included in nondegenerate_chars for "
+                            "class %s" % name)
+
+            if len(cls.gap_chars & cls.degenerate_chars) > 0:
+                raise GrammaredSequenceException(
+                    "gap_chars and degenerate_chars must not share any "
+                    "characters for class %s" % name)
+
+            if len(cls.gap_chars & cls.nondegenerate_chars) > 0:
+                raise GrammaredSequenceException(
+                    "gap_chars and nondegenerate_chars must not share any "
+                    "characters for class %s" % name)
+
+            if len(cls.degenerate_chars & cls.nondegenerate_chars) > 0:
+                raise GrammaredSequenceException(
+                    "degenerate_chars and nondegenerate_chars must not share "
+                    "any characters for class %s" % name)
+
+        return cls
+
+
+class DisableSubclassingMeta(GrammaredSequenceMeta):
+    def __new__(cls, name, bases, classdict):
+        for b in bases:
+            if isinstance(b, DisableSubclassingMeta):
+                raise TypeError("Subclassing disabled for class %s. To create"
+                                " a custom sequence class, inherit directly"
+                                " from skbio.sequence.%s" %
+                                (b.__name__, GrammaredSequence.__name__))
+        return type.__new__(cls, name, bases, dict(classdict))
```

Comment:
Should this be `super` so that inheritance works correctly? I'm not sure what the MRO is here, is it short-circuiting the `ABCMeta` and `GrammaredSequenceMeta` validations?